### ISSUE TITLE
No t.Foo.ok for all the tests

### ISSUE THREAD

**Issue by bsdimp** [onetrueawk/awk#39]
while I can test awk against the latest awk with regress, I get a few unexpected differences when I test again gawk. It's unclear to me which one is right: nawk or gawk in these cases. Is there a way to know, or is nawk right by definition?

**Comment by arnoldrobbins** [comment]
Thanks for the note.  The current test suite is something of a weak link, at least IMHO; it's the last item in the `TODO` file I made some time ago that isn't done.

That said, there are implementation differences between this awk and gawk; anything depending upon the random number generator or upon array traversal order, for instance, will be different between the two, but not necessarily "wrong". t.in2 is one example of such.

To answer your question in any more specific detail would require going through all the tests where there is a "failure" and figuring out why, which I don't really have the cycles to do right now.

For now, you can assume that "nawk is right by definition".  I hope one day to invest some time in the test suite, but I don't know when that will be.

**Comment by bsdimp** [comment]
I tried to pull together all the *BSD patches into one git repo in the summer of 2017 when it looked like the onetrueawk repo wasn't going to ever be updated. At least one of these patches broke things and life got busy at work so I let it drop....  I've started to dust it off again now that it appears I missed the project coming back to life.

I was hoping we could add all these tests to our CI process so that commits that broke awk would get flagged right away. But to do that I'd need the expected output. I can create that easily enough by hacking the Compare.* scripts to produce it and/or use it instead of $oldawk (likely by creating an oldawk script that just cats the output believed to be correct). But that assumes that FreeBSD's version of nawk has no bugs (not safe today, since it hasn't been updated with the many bug fixes since August 2017 that have gone in) so things could be generated properly (or that the Linux version has no bugs, etc).  Having it centralized in the repo would give a higher level of confidence, but I understand that might be difficult since the regression tests are contained inside a tarball...

I believe that you'd see the same deltas I am if you used gawk as oldawk, so I'm satisfied with your 'trust nawk'.
